_posts/2020-03-29-all-you-need-is-lambda-1-booleans.md

```diff
@@ -0,0 +1,260 @@
+
+    In Haskell, we would write `\ x -> y` instead; in JavaScript, `function (x) { return y }` or `(x) => y`.
+
+2. Function application (“lambda application,” or “application” or “function calls”).
+
+    Applications are written as `x y`, for expressions x and y, and left-associated, i.e. `a b c` = `(a b) c` ≠ `a (b c)`. Function application binds tighter than lambda abstraction, i.e. `λ x . λ y . y x` = `λ x . λ y . (y x)` ≠ `λ x . (λ y . y) x`.
+
+    The syntax is the same in Haskell; in JavaScript, we would write `x(y)` or `a(b, c)`. Note however that since lambda calculus functions are all single-argument functions, a more direct (tho less idiomatic) equivalent for the latter would be `a(b)(c)`.
+
+3. Variables (written as names; bound by enclosing lambdas; no `let` bindings, no globals; we will feel free to give variables names both great and small, and will include non-alphanumeric characters in names as we see fit, since the paucity of syntax means there’s little risk of ambiguity).
+
+Since a lambda is the only way to introduce a value—it’s the only “literal” syntax, and there are no predefined global definitions or standard library—we can also see that the only kinds of runtime values must be closures, perhaps consisting of the name of the introduced variable, body of the lambda, & values of any variables it closed over when made (again, we assume strict lexical scoping).
+
+How do you _do_ anything when you don’t even have `true` and `false`? Lambdas and variables don’t _do_, they merely _are_, so that leaves application. When all you have is lambda application, everything looks like a lambda abstraction, so we’ll represent booleans using lambdas.
+
+> Aside: Due to the lambda calculus’ terseness, I will be making free use of several notational conveniences:
+>
+> 1. writing `λ x y . z` as an abbreviation of `λ x . λ y . z`.
+> 2. writing `?` to stand for bits we don’t know yet, as tho we had an environment supporting holes.
+> 3. writing definitions as tho we had a metalanguage.
+> 4. referencing definitions elsewhere as tho we had globals.
+> 5. writing type signatures as tho we had a type system, and even a typechecker, with as much polymorphism and inference as is convenient at any particular moment.
+> 6. using syntactic recursion as tho it existed.
+> 7. using general recursion as tho it made sense.
+> 8. ignoring application order, normalization, reduction, substitution, values, references, allocation, copying, space, time, entropy, and any and all other such details whenever I feel like it.
+>
+> Further, I will by convention name types in `TitleCase` and both term and (local) type variables in `camelCase`.
+>
+> I will try to avoid pulling rabbits from hats too wantonly, but for now, I’ll ask you to suspend disbelief; I hope to revisit and justify some of these in later posts.
```

Comment:
I don't think you necessarily need this disclaimer. The above syntactic sugar is hardly excessive.

Reply:
I mostly want to indicate that I intend to justify even more reduction of syntax into the lambda calculus.